doc/documentation_workflow.rst

```diff
@@ -0,0 +1,154 @@
+
+.. code-block:: bash
+
+   conda deactivate
+   conda remove --name nest-doc --all
+
+Generating documentation with Sphinx
+++++++++++++++++++++++++++++++++++++
+
+Now that you activated your environment, you can generate HTML files using Sphinx.
+
+Sphinx uses reStructuredText. To learn more on how to use it, check out this `quick reference <https://docutils.sourceforge.io/docs/user/rst/quickref.html>`_.
+
+Editing and creating pages
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. You can edit and/or add ``.rst`` files in the ``doc`` directory using your editor of choice.
+
+2. If you create a new page, open ``contents.rst`` in the ``doc`` directory and add the file name under ``.. toctree::``. This will ensure it appears on the NEST simulator documentation's table of contents.
+
+3. If you rename or move a file, please make sure you update all the corresponding cross-references.
+
+4. Save your changes.
+
+Rendering HTML
+~~~~~~~~~~~~~~
+
+Using Sphinx, you can build documentation locally and preview it offline:
+
+1. Go to the ``doc`` folder in the source directory:
+
+.. code-block:: bash
+
+    cd </path/to/nest_source>/doc
+
+2. Generate HTML files:
+
+.. code-block:: bash
+
+   make html
+
+3. Preview files. They are then located in ``./docs/_build/html``:
+
+.. code-block:: bash
+
+   cd </path/to/nest_source>/doc/_build/html
+   browser filename.html
+
+4. If you add further changes to your files, repeat steps 1-3.
+
+Previewing on Read the Docs (optional)
+++++++++++++++++++++++++++++++++++++++
+
+Proceed as follows to preview your version of the documentation on Read the Docs.
+
+1. Check that unwanted directories are listed in ``.gitignore``:
+
+.. code-block:: bash
+
+   _build
+   _static
+   _templates
+
+2. Add, commit and push your changes to GitHub.
+
+3. Go to `Read the Docs <https://readthedocs.org/>`_. Sign up for an account if you don't have one.
+
+4. `Import <https://readthedocs.org/dashboard/import/>`_ the project.
+
+5. Enter the details of your project in the ``repo`` field and hit ``Create``.
+
+6. `Build your documentation <https://docs.readthedocs.io/en/stable/intro/import-guide.html#building-your-documentation>`_.
+
+This allows you to preview your work on your Read the Docs account. In order to see the changes on the official NEST simulator documentation, please submit a pull request (see below).
+
+Creating pull request
++++++++++++++++++++++
+
+When you feel your documentation work is finished, you can create a pull request to the ``master`` branch of the NEST Source Code Repository. Your pull request will be reviewed by our NEST Documentation Team.
+
+Developer documentation
++++++++++++++++++++++++
+
+For **developer documentation**, we use `doxygen <http://doxygen.org/>`__
+comments extensively throughout NEST.
+After installing NEST, you can extract comments from the source code with
+``make doc`` and a doxygen folder with html files will be generated in the doc
```

Comment:
`make doc` produces the following error for me:

➜ make doc
Running Sphinx v2.4.4
/bin/sh: doxygen: command not found
Wrote model documentation for model models/multimeter
Wrote model documentation for model models/spike_detector
Wrote model documentation for model models/weight_recorder
Wrote model documentation for model nestkernel/recording_backend_ascii
Wrote model documentation for model nestkernel/recording_backend_memory
Wrote model documentation for model nestkernel/recording_backend_screen
Wrote model documentation for model nestkernel/recording_backend_sionlib
/Users/work/opt/miniconda3/envs/nest-doc/lib/python3.8/site-packages/sphinx/util/docutils.py:285: RemovedInSphinx30Warning: function based directive support is now deprecated. Use class based directive instead.
  warnings.warn('function based directive support is now deprecated. '

Sphinx error:
Builder name doc not registered or available through entry point
make: *** [doc] Error 2